.github/workflows/sign-nuget-package.yaml

```diff
@@ -0,0 +1,143 @@
+name: Sign NuGet Package
+
+on:
+  workflow_run:
+    workflows: [Build My Package]
+    types: [completed]
+    branches: [main, release]
+
+permissions:
+  contents: read
+  actions: read
+
+jobs:
+  sign:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        continue-on-error: true
+        with:
+          name: nuget-package
+          path: artifacts
+
+      - name: Check if artifacts were downloaded
+        id: check-artifacts
+        run: |
+          if [ -d "artifacts" ] && [ "$(ls -A artifacts 2>/dev/null)" ]; then
+            echo "artifacts_found=true" >> $GITHUB_OUTPUT
+            echo "✅ Artifacts found and downloaded successfully"
+          else
+            echo "artifacts_found=false" >> $GITHUB_OUTPUT
+            echo "⚠️ No artifacts found, attempting to build locally"
+          fi
+
+      - name: Build locally if no artifacts found
```

Comment:
I think we may be stepping outside of our bounds by building artifacts in this step. We should really just be signing artifacts here.